src/controllers/queries/index.ts

```diff
@@ -401,120 +373,147 @@ export class QueriesController extends HTMLController {
     return a.length
   }
 
-  /**
-   *
-   * @param rpcResponse
-   * @param log - logger, passed so we can keep track of req_id
-   * @param method - either receive or sent needed for submitting drpc request
-   * @param payload - rest(spread operator) that will contain
-   * @returns html either success of error
-   */
-  public async submitDrpcRequest({
-    rpcResponse,
-    log,
-    method,
-    ...payload
-  }: {
-    query: QueryRow | null
-    method: 'submit_query_request' | 'submit_query_response'
-    log: Logger
-    localQuery: { query: string; queryIdForResponse?: UUID; emissions?: string; quantity?: number; productId?: string }
-    rpcResponse?: DrpcResponse | undefined
-    parentId?: UUID | undefined
-    connectionId?: UUID
-    connection?: ConnectionRow | null
-  }) {
-    try {
-      const { parentId, connection, localQuery, connectionId } = payload
-      const [conn]: ConnectionRow[] = connection
-        ? [connection]
-        : await this.db.get('connection', { id: connectionId, status: 'verified_both' }, [['updated_at', 'desc']])
-
-      if (!conn.agent_connection_id) throw new InvalidInputError('missing agent_id from connection')
-      if (parentId) {
-        const [query] = await this.db.insert('query', {
-          connection_id: conn.id,
-          parent_id: parentId,
-          query_type: 'Scope 3 Carbon Consumption',
-          status: 'pending_their_input',
-          details: localQuery,
-          response_id: null,
-          query_response: null,
-          role: 'requester',
-        })
-        if (!payload.query) payload.query = query
+  private async handleQueryRequest(
+    log: Logger,
```

Comment:
@dblane-digicatapult say we need more logging

Reply:
agreed next PR I'll fill in the gaps